indexer/src/sinks/cardano.rs

```diff
@@ -274,20 +301,52 @@ async fn insert_block(
     txn: &DatabaseTransaction,
     exec_plan: Arc<ExecutionPlan>,
     task_perf_aggregator: Arc<Mutex<TaskPerfAggregator>>,
-) -> Result<(), DbErr> {
+    previous_era: Option<EraValue>,
+    custom_genesis_folder: Option<String>,
+    network: String,
+) -> Result<EraValue, DbErr> {
     let mut perf_aggregator = PerfAggregator::new();
 
     let block_parse_counter = std::time::Instant::now();
 
     let block_payload = hex::decode(cbor_hex.clone()).unwrap();
     let multi_block = MultiEraBlock::from_explicit_network_cbor_bytes(&block_payload).unwrap();
 
+    let era = to_era_value(&multi_block);
     let block_global_info = BlockGlobalInfo {
-        era: to_era_value(&multi_block),
+        era,
         epoch,
         epoch_slot,
     };
 
+    match previous_era {
+        None => {
+            let genesis_file = get_genesis_file(&network, &custom_genesis_folder, "byron");
+            tasks::genesis::genesis_executor::process_genesis_block(
+                txn,
+                ("", &genesis_file, &block_global_info),
+                &exec_plan,
+                task_perf_aggregator.clone(),
+            )
+            .await?
+        }
+        Some(prev_era) if era > prev_era => {
```

Comment:
IIRC I never added proper rollback support since it was tricky to figure out how to handle rollbacks that undo a era transition